src/group_theory/subgroup/basic.lean

```diff
@@ -1476,6 +1476,10 @@ iff.rfl
 @[to_additive] lemma range_eq_map (f : G →* N) : f.range = (⊤ : subgroup G).map f :=
 by ext; simp
 
+@[to_additive]
+lemma range_eq_top_of_surjective {f : G →* N} (hf : function.surjective f) :
+  f.range = ⊤ := by { ext, simp, apply hf }
```

Comment:
Incidentally the `simp` here is an example of a [non-terminal simp](https://leanprover-community.github.io/extras/simp.html#non-terminal-simps) something which we avoid.